src/redux/sagas/UserProfileSaga/UserProfilePatchSaga.js

```diff
@@ -0,0 +1,34 @@
+import { call, put, takeEvery } from 'redux-saga/effects';
+import * as types from '../../actions/userProfileActions/types';
+import { api } from '../../../utils/auth';
+
+
+export const apiClient = {
+  updateProfile: (username, data) => {
+    const url = `/profiles/${username}/`;
+    return api.patch(url, data);
+  },
+};
+export function* userProfilePatchSaga({ payload }) {
+    const { username, data } = payload;
+  try {
+    const response = yield call(apiClient.updateProfile, username, data);
+    yield put({
+      type: types.PROFILE_PATCH_SUCCESS,
+      payload: response.data,
+    });
+  } catch (err) {
+    yield put({
+      type: types.PROFILE_PATCH_ERROR,
+      payload: { errors: err.response },
+    });
+  }
+}
+
+function* watchuserProfilePatch() {
+  yield takeEvery(
```

Comment:
Why did you choose to use `takeEvery` and not `takeLatest`?

Reply:
Hi,
I was considering the user might want to send several requests.
When this happens the application gets to deal with the latest request.
Otherwise no other particular reason.
You may advise me on which is most appropriate, thank you.

Regards
Steven Ennis